src/test_functions/test_functions.rs

```diff
@@ -0,0 +1,507 @@
+use std::f64;
+
+
+/// # Rastrigin function
+/// n-dimensional \
+/// Global minimum: \
+/// f(0, ..., 0) = 0
+pub fn rastrigin(x: &Vec<f64>) -> f64 {
+    if x.len() < 1 {
+        panic!("Rastrigin function takes an at least one dimensional vector as a parameter.");
+    }
+    let mut result: f64 = 0.0;
+    for x_curr in x.iter() {
```

Comment:
I believe using `.iter()` explicitly here is not necessary. Compiler will deduce this on its own. 
This can be applied in few more places.